src/main/java/org/gradle/wrapper/GroovyWrapperMain.java

```diff
@@ -55,13 +54,23 @@ public static void main(String[] args) throws Exception {
 
         addSystemProperties(gradleUserHome, rootDir);
 
-        Logger logger = logger(options);
+        WrapperConfiguration config = WrapperExecutor.forWrapperPropertiesFile(propertiesFile).getConfiguration();
+        PathAssembler.LocalDistribution localDist = new PathAssembler(gradleUserHome).getDistribution(config);
+        File gradleHome = getAndVerifyDistributionRoot(localDist.getDistributionDir());
 
-        WrapperExecutor wrapperExecutor = WrapperExecutor.forWrapperPropertiesFile(propertiesFile);
-        wrapperExecutor.execute(
-                args,
-                new Install(logger, new Download(logger, "gradlew", wrapperVersion()), new PathAssembler(gradleUserHome)),
-                new GroovyBootstrapMainStarter());
+        new GroovyBootstrapMainStarter().start(args, gradleHome);
+    }
+
+    static File getAndVerifyDistributionRoot(File distDir) {
+        // Extracted from org.gradle.wrapper.Install.getAndVerifyDistributionRoot
```

Comment:
and then if you *keep* that `Install` invocation, there's no reason to have this duplicate copy of this method brought over from the Install class.